File: gaiag/baseline/component/AlarmSystem.java
// Gaiag --- Guile in Asd In Asd in Guile.
//
// This file is part of Gaiag.
//
// Copyright © 2014 Jan Nieuwenhuizen <janneke@gnu.org>
//
// Gaiag is free software: you can redistribute it and/or modify it
// under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// Gaiag is distributed in the hope that it will be useful, but
// WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
// Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public
// License along with Gaiag.  If not, see <http://www.gnu.org/licenses/>.
//
// Commentary:
//
// Code:

class AlarmSystem {
  Alarm alarm;
  Sensor sensor;
  Siren siren;
  IConsole console;

  public AlarmSystem() {
    alarm = new Alarm();
    sensor = new Sensor();
    siren = new Siren();
    console = alarm.console;

    Interface.connect(sensor.sensor, alarm.sensor);
    Interface.connect(siren.siren, alarm.siren);
  };
}

File: gaiag/baseline/AlarmSystem.js
// Dezyne --- Dezyne command line tools
//
// Copyright © 2014 Jan Nieuwenhuizen <janneke@gnu.org>
//
// This file is part of Dezyne.
//
// Dezyne is free software: you can redistribute it and/or modify it
// under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// Dezyne is distributed in the hope that it will be useful, but
// WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
// Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public
// License along with Dezyne.  If not, see <http://www.gnu.org/licenses/>.
//
// Commentary:
//
// Code:

dezyne.AlarmSystem = function() {
  this.alarm = new dezyne.Alarm();
  this.sensor = new dezyne.Sensor();
  this.siren = new dezyne.Siren();
  this.console = this.alarm.console;

  dezyne.connect(this.sensor.sensor, this.alarm.sensor);
  dezyne.connect(this.siren.siren, this.alarm.siren);

};
